### PR TITLE
Questionnarie table added to the database

### PR DESCRIPTION
**Pull request by siddh1004**
https://app.shortcut.com/simpledotorg/story/9557/add-sync-resource-for-questionnaire-layout-api

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/monthlyReports/Questionnaire.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import kotlinx.parcelize.Parcelize
+import java.util.UUID
+
+@Entity(tableName = "Questionnaire")
+@Parcelize
+data class Questionnaire(
+    @PrimaryKey
+    val uuid: UUID,
+
+    val questionnaire_type: QuestionnaireType,
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/monthlyReports/Questionnaire.kt:15]: I think we discussed of making `QuestionnaireType` as primary key in PRD. Please check

File: app/src/main/java/org/simple/clinic/monthlyReports/QuestionnaireType.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import com.squareup.moshi.ToJson
+import kotlinx.parcelize.Parcelize
+import org.simple.clinic.util.room.SafeEnumTypeAdapter
+
+sealed class QuestionnaireType : Parcelable {
+
+  object TypeAdapter : SafeEnumTypeAdapter<QuestionnaireType>(
+      knownMappings = mapOf(
+          MonthlyScreeningReports to "monthly_screening_reports",
+          MonthlyDrugStock to "monthly_drug_stock"
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/monthlyReports/QuestionnaireType.kt:15]: Is this type already being provided by the server? If not please remove this. We can add it once server has actual implementation of it.

**Review by msasikanth — CHANGES_REQUESTED** [review]
Looks good overall. Left couple of comments, PTAL

File: app/src/main/java/org/simple/clinic/monthlyReports/Questionnaire.kt (hunk 1 above)
Comment (siddh1004) [reply app/src/main/java/org/simple/clinic/monthlyReports/Questionnaire.kt]: Primary key update to `questionnaire_type`

File: app/src/main/java/org/simple/clinic/monthlyReports/QuestionnaireType.kt (hunk 2 above)
Comment (siddh1004) [reply app/src/main/java/org/simple/clinic/monthlyReports/QuestionnaireType.kt]: Data type of `questionnaire_type` changed to string and `QuestionnaireType.kt` deleted

Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/monthlyReports/QuestionnaireType.kt]: We would still need the questionnaire type class, I am just saying we don't need the drug stock type. Right now it's just screening reports.

Comment (siddh1004) [reply app/src/main/java/org/simple/clinic/monthlyReports/QuestionnaireType.kt]: I have again added the questionnaire type and removed the monthly drug stock type.

**Pull request merged**